src/main/java/Duke.java

```diff
@@ -1,10 +1,406 @@
+import exceptions.EmptyDescriptionException;
+import exceptions.EmptyNumberException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    private static final ArrayList<Task> taskList = new ArrayList<>();
+
+    // These are the prefix strings to define the data type of a command parameter
+    private static final String COMMAND_TODO_WORD = "todo";
+    private static final String COMMAND_EVENT_WORD = "event";
+    private static final String COMMAND_DEADLINE_WORD = "deadline";
+    private static final String COMMAND_LIST_WORD = "list";
+    private static final String COMMAND_EXIT_WORD = "bye";
+    private static final String COMMAND_DONE_WORD = "done";
+    private static final String COMMAND_DELETE_WORD = "delete";
+
+    // error messages
+    private static final String MESSAGE_DESCRIPTION_EMPTY_TODO = "The description of a todo cannot be empty.";
+    private static final String MESSAGE_DESCRIPTION_EMPTY_DEADLINE = "The description of a deadline cannot be empty.";
+    private static final String MESSAGE_DESCRIPTION_EMPTY_EVENT = "The description of a event cannot be empty.";
+    private static final String MESSAGE_DESCRIPTION_EMPTY_DONE = "The number of the task to be marked as done cannot be empty.";
+    private static final String MESSAGE_DESCRIPTION_EMPTY_DELETE = "The number of the task to be deleted cannot be empty.";
+    private static final String MESSAGE_INVALID_COMMAND = "I'm sorry, but I don't know what that means.";
+    private static final String MESSAGE_INVALID_DEADLINE = "No deadline provided or wrong splitter.";
+    private static final String MESSAGE_INVALID_EVENT = "No event date and time provided or wrong splitter.";
+    private static final String MESSAGE_INVALID_COMMAND_DONE = "Number not provided for the task to be marked as done.";
+    private static final String MESSAGE_INVALID_NUMBER_DONE = "Task number does not exist.";
+    private static final String MESSAGE_INVALID_COMMAND_DELETE = "Number not provided for the task to be deleted.";
+    private static final String MESSAGE_INVALID_NUMBER_DELETE = "Task number does not exist.";
+    private static final String MESSAGE_SAVE_FILE_LOADED = "Save file loaded.";
+    private static final String MESSAGE_SAVE_FILE_CREATED = "Save file created.";
+    private static final String MESSAGE_SAVE_FILE_SAVED = "File saved.";
```

Comment:
Good job on extracting the messages out, you may want to create a `Message` class that stores all messages